Fix some layout issues and refactor js code

issammani opened the pull request:
This PR fixes some layout issues mainly:

- Changes font-size from pt to px as it's not recommended. [[ Source ]](https://www.w3.org/Style/Examples/007/units.en.html) 
- Fixes table and image overflow on smaller screens.
- Removes center block since its obsolete. [[ Source ]](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/center#:~:text=The%20obsolete%20HTML%20Center%20Element,to%20be%2C%20.)
- Fixes image aspect ratio problem and resizes big images.

This PR also introduces some JS changes mainly:

- Removes gen_mail_to_link function.
- Moves the addDownloadLink into a new js file.
- Refactors addDownloadLink.

issammani committed:
Change font-size unit from pt to px

- the pt unit is not recommended for screen use.
  [https://www.w3.org/Style/Examples/007/units.en.html]

issammani committed:
Fix navbar layout

- Increase navbar collapse breakpoint to avoid nav items breaking to a new line.

- Decrease font size of nav items.

- Increase clickable target of nav items.

issammani committed:
Remove center block around logo

- Replace center block with a flex container. Center blocks are
  obsolete.
  [Source: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/center#:~:text=The%20obsolete%20HTML%20Center%20Element,to%20be%2C%20.]

- Add margin bottom to logo.

issammani committed:
Change gold color

- The previous color had a low contrast ratio.

issammani committed:
Add margin to relevant links buttons

- Add margin bottom and right.

- Remove style attribute and use bootstrap margin spacing classes.

issammani committed:
Wrap responsive-table around tables

- This fixes tables overflowing in small screens.

- Removed responsive-table, where misused. The purpose responsive-table container is to wrap around a table.
  Using it as a top level container instead of col caused the whole column to scroll horizentally instead of just the table.

issammani committed:
Move favicon link to head

issammani committed:
Fix overflow issue in intro section

- Add column wrappers.

- Properly size images.

issammani committed:
Remove redundant favicon files

issammani committed:
Fix team images issues

- Download all images locally.

- Resize big images to 450x450 ( Enough since images have a fixed width
  of 150x150).

- Change aspect ratio of Corby Rosset image to 1:1.

issammani committed:
Remove gen_mail_to_link and encode mailto directly

- The usage of document.wrirte is considered bad practice.

- Save javascript parsing time.

issammani committed:
Move addDownloadLink to its own js file

ghost commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/microsoft/msmarco?pullRequest=1) <br/>All CLA requirements met.

issammani committed:
Refactor addDownloadLinks function

- Add datasets object to hold all necessary information to render
  datasets.

- Add click event listener in js instead of html's onclick tag.

- Add linksShown flag to prevent useless re-rendering of datasets after
  first render.
